dgl_ptm/dgl_ptm/agentInteraction/trade_money.py

```diff
@@ -30,14 +30,14 @@ def trade_money(agent_graph, method: str):
     
     # Transfer of wealth
     if method == 'weighted_transfer':
-        _weighted_transfer(agent_graph)
+        _weighted_transfer(agent_graph, device)
     elif method == 'singular_transfer':
-        _singular_transfer(agent_graph)
+        _singular_transfer(agent_graph, device)
     else:
         raise NotImplementedError("Incorrect method received. \
                          Method needs to be either 'weighted_transfer' or 'singular_transfer'")
     
-def _weighted_transfer(agent_graph):
+def _weighted_transfer(agent_graph, device):
```

Comment:
The argument device is added but it is not used inside the function `_weighted_transfer`.

Reply:
This was a bug. Its fixed now.